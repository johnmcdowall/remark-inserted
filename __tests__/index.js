const dedent = require("dedent");
const unified = require("unified");
const reParse = require("remark-parse");
const remarkStringify = require("remark-stringify");
const rehypeStringify = require("rehype-stringify");
const remark2rehype = require("remark-rehype");

const plugin = require("../src/");

const render = text =>
  unified()
    .use(reParse)
    .use(plugin)
    .use(remark2rehype)
    .use(rehypeStringify)
    .processSync(text);

const markdown = dedent`
 ++cats are awesome++
`;

describe("parses inserted markdown", () => {
  it("generates the correct mark HTML elements", () => {
    const { contents } = render(markdown);
    expect(contents).toMatchSnapshot();
  });
});

test("to markdown", () => {
  const { contents } = unified()
    .use(reParse)
    .use(remarkStringify)
    .use(plugin)
    .processSync(markdown);

  expect(contents).toMatchSnapshot();
});
