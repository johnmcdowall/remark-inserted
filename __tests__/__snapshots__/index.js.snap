// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses inserted markdown generates the correct mark HTML elements 1`] = `"<p><mark>cats are awesome</mark></p>"`;

exports[`to markdown 1`] = `
"++cats are awesome++
"
`;
